### PR TITLE
Make PhotoSubjects use a one to many relationship

### PR DESCRIPTION
**Pull request by Slendy**
This PR changes the Photo and PhotoSubject system to use a one (photo) to many (photosubjects) relationship. The benefits of this are increased query performance and better constraints because it's impossible to have a PhotoSubject with no parent photo. The migration assigns PhotoId's to existing PhotoSubjects by using the MySQL `find_in_set` function which can properly compare the PhotoSubjectId and the entry in the comma-delimited list. ~~In my testing, the migration took over 30 seconds which exceeded the command timeout for EntityFramework and caused the default docker healthcheck to fail which results in an incomplete migration.~~ 
The timeout issue and docker health check issues have both been fixed.

### CODE REVIEW

**Review by daggintosh — APPROVED** [review]
I love Collections! An excellent optimization. Lighthouse just got much faster!

**Comment by sudokoko** [comment]
had to re-teach him the way
Screen Shot 2023-02-23 at 10 09 38 PM

File: ProjectLighthouse/Types/Entities/Profile/Photo.cs [hunk 1]
```diff
@@ -34,17 +32,14 @@ public class PhotoSlot
...
-    private List<PhotoSubject>? _subjects;
-
     [NotMapped]
     [XmlElement("slot")]
     public PhotoSlot? XmlLevelInfo;
 
     [NotMapped]
     [XmlArray("subjects")]
     [XmlArrayItem("subject")]
-    public List<PhotoSubject>? SubjectsXmlDontUseLiterallyEver;
```
Comment (Zaprit) [inline ProjectLighthouse/Types/Entities/Profile/Photo.cs:47]: lol

**Review by Zaprit — COMMENTED** [review]
LGTM

**Comment by Zaprit** [comment]
yes i know i'm late

**Pull request merged**